draft! Add linter

demoneno4ec opened the pull request:
- fix tests
- and fix lint problems

#10

demoneno4ec commented:
Донастроить revive
- [x] add-constant
- [x] argument-limit
- [x] cognitive-complexity
- [x] confusing-naming
- [x] context-as-argument
- [x] cyclomatic
- [x] empty-lines
- [ ] file-header
- [ ] flag-parameter
- [ ] function-result-limit
- [ ] function-length
- [x] if-return
- [x] import-shadowing
- [x] line-length-limit
- [x] max-public-structs
- [x] modifies-parameter
- [x] nested-structs
- [x] optimize-operands-order
- [x] var-naming
- [x] var-declaration
- [x] unhandled-error

demoneno4ec commented:
Добавить конфиги для оставшихся линтеров

- [x] [gosec](https://golangci-lint.run/usage/linters/#gosec)
- [x] [gocritic](https://golangci-lint.run/usage/linters/#gocritic)
- [x] [stylecheck](https://golangci-lint.run/usage/linters/#stylecheck)

rusinikita commented:
Договорились на такой конфиг

```
linters-settings:
  gosimple:
    checks: [ "all" ]
  govet:
    check-shadowing: true
    enable-all: true
    disable:
      - fieldalignment
  staticcheck:
    checks: [ "all" ]
  revive:
    ignore-generated-header: true
    enable-all-rules: true
    rules:
      - name: add-constant
        arguments:
          - maxLitCount: "3"
            allowInts: "0,1"
      - name: argument-limit
        arguments: [ 4 ]
      - name: banned-characters
        disabled: true
      - name: cognitive-complexity
        arguments: [ 8 ]
      - name: context-as-argument
        arguments:
          - allowTypesBefore: "*testing.T"
      - name: cyclomatic
        arguments: [ 7 ]
      - name: defer
        arguments:
          - [ "call-chain", "loop" ]
      - name: file-header
        disabled: true
      - name: function-result-limit
        arguments: [ 2 ]
      - name: function-length
        arguments: [ 10, 20 ]
      - name: line-length-limit
        arguments: [ 80 ]
      - name: max-public-structs
        arguments: [ 3 ]
      - name: string-format
        arguments:
          - - 'core.WriteError[1].Message'
            - '/^([^A-Z]|$)/'
            - must not start with a capital letter
          - - 'fmt.Errorf[0]'
            - '/(^|[^\.!?])$/'
            - must not end in punctuation
          - - panic
            - '/^[^\n]*$/'
            - must not contain line breaks
      - name: unhandled-error
        arguments:
          - "fmt.Printf"
  stylecheck:
    checks: [ "all" ]
  gosec:
    exclude-generated: true
    severity: high
    concurrency: 3
  gocritic:
    enabled-checks:
      - hugeParam
      - tooManyResultsChecker
      - rangeValCopy
      - rangeExprCopy
      - nestingReduce
      - unnamedResult
      - truncateCmp
      - ruleguard
      - elseif
      - captLocal
      - underef
    settings:
      captLocal:
        paramsOnly: false
      elseif:
        skipBalanced: false
      hugeParam:
        sizeThreshold: 256
      truncateCmp:
        skipArchDependent: false
      underef:
        skipRecvDeref: false
      unnamedResult:
        checkExported: true
linters:
  disable-all: true
  enable:
    - errcheck
    - gosimple
    - govet
    - ineffassign
    - staticcheck
    - unused
    - revive
    - stylecheck
    - gosec
    - gocritic
issues:
  # See the dedicated "issues" documentation section.
  exclude-rules:
    - path: _test\.go
      text: "add-constant:"

```

rusinikita commented on dashboard/file.go:
```diff
@@ -0,0 +1,40 @@
+	"github.com/go-echarts/go-echarts/v2/opts"
+)
+
+type file struct {
```
Верни, пожалуйста, конструктор. if внутри можно удалить

Commit:
Add linter config

- add linter config
- enable linters
- errcheck
    - gosimple
    - govet
    - ineffassign
    - staticcheck
    - unused
    - revive
    - stylecheck
    - gosec
    - gocritic

#10

Commit:
Delete search folder

- delete search folder with some lint problems

#10

Commit:
Fix problems

- fix linter problems

#10
